.github/workflows/test-google-cloud-platform-install.yaml

```diff
@@ -0,0 +1,204 @@
+          --machine-type e2-medium \
+          --num-nodes 2
+
+        # note: num-nodes will be created in each zone, such that if you specify
+        # --num-nodes=4 and choose two locations 8 nodes will be created.
+
+    - name: Create new GCP global static IP address
+      id: static_ip_creation
+      # note: we need to create the IP address first in order
+      # to get the load balancer successfully provisioned
+      run: |
+        gcloud compute addresses create \
+          --project ${{ secrets.GCP_PROJECT_ID }} \
+          --global \
+          ${{ steps.vars.outputs.dns_record }}
+
+    - name: Create the DNS record
+      id: dns_creation
+      run: |
+        # Get the global static IP address
+        global_static_ip=$(gcloud compute addresses list --project ${{ secrets.GCP_PROJECT_ID }} --global --filter=name:${{ steps.vars.outputs.dns_record }} --format="value(ADDRESS)")
+
+        # Create the DNS record
+        doctl compute domain records create \
+          posthog.cc \
+          --record-type A \
+          --record-ttl 60 \
+          --record-name "${{ steps.vars.outputs.dns_record }}" \
+          --record-data "$global_static_ip"
+
+    - name: Install PostHog using the Helm chart
+      run: |
+        helm upgrade --install \
+          -f ci/values/google_cloud_platform.yaml \
+          --set "ingress.hostname=${{ steps.vars.outputs.fqdn_record }}" \
+          --set "ingress.gcp.ip_name=${{ steps.vars.outputs.dns_record }}" \
+          --timeout 20m \
+          --create-namespace \
+          --namespace posthog \
+          posthog ./charts/posthog \
+          --wait-for-jobs \
+          --wait
+
+    #
+    # Wait for all k8s resources to be ready.
+    #
+    # Despite the --wait flag used in the command above
+    # there is no guarantee that all the resources will be deployed
+    # when the command returns.
+    #
+    #
+    # Why can't we directly use the 'action-k8s-await-workloads' step below?
+    # Because it's not working for this use case
+    #
+    # ref: https://github.com/jupyterhub/action-k8s-await-workloads/issues/38
+    #
+    - name: Workaround - wait for all the k8s resources to be ready
+      timeout-minutes: 15
+      run: |
+        echo "Waiting for pods to be ready..."
+        while ! kubectl wait --for=condition=Ready pods --timeout=60s --all -n posthog > /dev/null 2>&1
+        do
+          echo "  pods are not yet ready"
+        done
+        echo "All pods are now ready!"
+
+        echo "Waiting for the GCP Load Balancer to be ready..."
+        load_balancer_external_ip=""
+        while [ -z "$load_balancer_external_ip" ];
+        do
+          echo "  sleeping 10 seconds" && sleep 10
+          load_balancer_external_ip=$(kubectl get ingress -n posthog posthog -o jsonpath="{.status.loadBalancer.ingress[0].ip}")
+        done
+        echo "The GCP Load Balancer is now ready!"
+
+    - name: Wait until all the resources are fully deployed in k8s
```

Comment:
Could we use sth like https://github.blog/changelog/2021-08-25-github-actions-reduce-duplication-with-action-composition/ to reduce the code duplication

Reply:
Yes and that's part of the plan. I'm going to do that once all the steps are fully fleshes on all platforms